packages/app-builder-lib/src/binDownload.ts

```diff
@@ -51,7 +51,7 @@ export function getBin(cacheKey: string, url?: string | null, checksum?: string
     return promise
   }
 
-  promise = doGetBin(cacheName, url, checksum)
+  promise = doGetBin(cacheKey, url, checksum)
```

Comment:
I don't think this will work as intended actually.

If the flow is as following:
1) `ELECTRON_BUILDER_CACHE` is set during first build
2) rebuild another app instance without `ELECTRON_BUILDER_CACHE` set

The "bin"/toolset used in step 2 would be the same custom toolset that was configured in step 1 due to cache key reuse.

Maybe we can use `${cacheKey}${process.env.ELECTRON_BUILDER_CACHE ?? ""}` instead?

Reply:
The cache managed by versionToPromise is only in memory and is therefore not reused by other instances.
I don't know exactly what its purpose is. I suppose it is used to manage parallelism of app-builder calls from the same instance.

Reply:
> I do not understand why this promises cache key is bound to ELECTRON_BUILDER_CACHE.

I don't understand this. It's a reliable key to use in a map that distinguishes a custom tooling cache from the default cache.

I see no reason why we aren't taking this approach:
> we can use `const cacheName = ${cacheKey}${process.env.ELECTRON_BUILDER_CACHE ?? ""}` instead

Reply:
What if we just did `path.basename(process.env.ELECTRON_BUILDER_CACHE ?? "")${cacheKey}`

I'm pretty sure that the custom cache path needs to be somehow leveraged to differentiate from the on-demand toolset versus the ELECTRON_BUILDER_CACHE toolset.

Reply:
Yeah that will probably work, but in the best case all it will do is get back the value of `cacheKey`, so why bother with the extra steps?
I am only 90% sure that will behave correctly in my case, as remember that `app-builder-bin` is ending up with a folder called `Cactions-runner-bitfocus_workcompanion-satellitecompanion-satellite.cacheelectron-buildernsis-3.0.4.1-nsis-3.0.4.1`. I expect that is a problem with escaping in the shell command, but am not 100% certain. 

To clarify, I'm advocating for this PR as is.  
I think what you are saying about the promise cache makes sense, and is a reason to keep the current `cacheName` behaviour for that. The call to `getBin` is the only thing that needs changing.